nimble/host/src/ble_eddystone.c

```diff
@@ -175,3 +176,44 @@ ble_eddystone_set_adv_data_url(struct ble_hs_adv_fields *adv_fields,
 
     return 0;
 }
+
+int
+ble_eddystone_set_adv_data_tlm_unencrypted(struct ble_hs_adv_fields *adv_fields,
+                               uint16_t battery_voltage, float temperature,
+                               uint32_t adv_pdu_count, uint32_t time_since_power_up)
+{
+    uint8_t *svc_data;
+    uint8_t temp_data[4];
+    int tmp;
+    int rc;
+
+    svc_data = ble_eddystone_set_svc_data_base(BLE_EDDYSTONE_FRAME_TYPE_TLM_UNENCRYPTED);
+
+    /* First byte is TLM Version - 00 for Unencrypted. */
+    svc_data[0] = 00;
+
+    /* Second paramter is Battery Voltage in mV. */
+    put_be16(temp_data, battery_voltage);
+    memcpy(svc_data + 1, temp_data, 2);
```

Comment:
you can write data directly to destination buffer, not need for extra copy

Reply:
Understood, will change it.